packages/autocomplete/src/containers/AutocompleteContainer.js

```diff
@@ -274,7 +274,7 @@ class AutocompleteContainer extends ControlledComponent {
       if (isOpen) {
         if (this.focusSelectionModel.hasSelection()) {
           this.selectItem(this.indexKeyMap[this.focusSelectionModel.selectedIndex]);
-        } else if (Object.keys(this.indexKeyMap).length > 0) {
+        } else if (Object.keys(this.indexKeyMap).length > 0 && e.target.value) {
```

Comment:
You're right, this should probably be handled as well.

I could see this being a necessity for some auto completes, but not others. Maybe this logic should be handled by the consumer in the `onSelect` handler? Might be something we have the examples fix?

Reply:
👍 I feel this is application logic. That's why I would take a step further and propose that this component does nothing about this piece of logic.

However, since it is done this way, let's try this and see. 😄